pkg/resource/trail/hooks.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package trail
+
+import (
+	"context"
+	"fmt"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	ackutil "github.com/aws-controllers-k8s/runtime/pkg/util"
+	svcsdk "github.com/aws/aws-sdk-go/service/cloudtrail"
+
+	svcapitypes "github.com/aws-controllers-k8s/cloudtrail-controller/apis/v1alpha1"
+)
+
+// setResourceAdditionalFields will describe the fields that are not return by
+// GetTrail calls
+func (rm *resourceManager) setResourceAdditionalFields(
+	ctx context.Context,
+	ko *svcapitypes.Trail,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.setResourceAdditionalFields")
+	defer exit(err)
+
+	// Set trail tags
+	ko.Spec.Tags, err = rm.getTrailTags(ctx, string(*ko.Status.ACKResourceMetadata.ARN))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getTrailTags retrieves a resource list of tags.
+func (rm *resourceManager) getTrailTags(ctx context.Context, resourceARN string) ([]*svcapitypes.Tag, error) {
```

Comment:
nit: please split long function signatures' parameters on separate lines, like you did above for `setResourceAdditionalFields`, so do like this:

```go
func (rm *resourceManager) getTrailTags(
    ctx context.Context,
    resourceARN string,
) ([]*svcapitypes.Tag, error) {
```